src/components/TodoAllChecked.tsx

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useContext, useState, useEffect } from 'react';
+import { TodoContext } from './TodoContext';
+import '../styles/todoapp.scss';
+
+export const TodoAllChecked: React.FC = () => {
+  const { todos, setTodos } = useContext(TodoContext);
+  const [isChecked, setIsChecked] = useState(
+    todos.every((todo) => todo.completed),
+  );
+
+  useEffect(() => {
+    if (todos.every((todo) => todo.completed)) {
+      setIsChecked(true);
+    } else {
+      setIsChecked(false);
+    }
+  }, [todos]);
```

Comment:
looks like redundant to make state for isChecked.
you always can check it with every method.
wnenever todos change TodoAllChecked will be rerender.
const isChecked = todos.every((todo) => todo.completed);